src/go/configgenerator/route_generator.go

```diff
@@ -367,21 +415,25 @@ func makeHttpExactPathRouteMatcher(path string) *routepb.RouteMatch {
 	}
 }
 
-func makeHttpRouteMatchers(httpRule *httppattern.Pattern) ([]*routepb.RouteMatch, error) {
+func makeHttpRouteMatchers(httpRule *httppattern.Pattern, seenUriTemplatesInRoute map[string]bool) ([]*routepb.RouteMatch, []*routepb.RouteMatch, error) {
 	if httpRule == nil {
-		return nil, fmt.Errorf("httpRule is nil")
+		return nil, nil, fmt.Errorf("httpRule is nil")
 	}
 	var routeMatchers []*routepb.RouteMatch
+	var uriTemplates []string
 
 	if httpRule.UriTemplate.IsExactMatch() {
 		pathNoTrailingSlash := httpRule.UriTemplate.ExactMatchString(false)
 		pathWithTrailingSlash := httpRule.UriTemplate.ExactMatchString(true)
 
+		uriTemplates = append(uriTemplates, pathNoTrailingSlash)
```

Comment:
This is very buggy code, it relies on uriTemplate array align with routeMatchers array.   Please refractory this.

Reply:
Done by wrapping them into a struct